tools/KeyfileIOUtils.cpp

```diff
@@ -0,0 +1,728 @@
+    } else {
+      std::cout << filename << ": line " << lineNumbers.at(identifier)
+        << ": expected integer value for " << identifier << ", found list.\n";
+    }
+    return false;
+  }
+
+  std::string strVal = map[identifier];
+  long long unvalidatedVal;
+
+  std::string errorMessage = filename + ": line "
+    + std::to_string(lineNumbers.at(identifier)) + ": Invalid value given for "
+    + identifier + ": " + strVal + " ( expected integer in range ["
+    + std::to_string(min) + ", " + std::to_string(max) + "], inclusive.\n";
+
+  try {
+    unvalidatedVal = std::stoll(strVal);
+  } catch (std::invalid_argument e) {
+    std::cout << errorMessage;
+    return false;
+  } catch (std::out_of_range e) {
+    std::cout << errorMessage;
+    return false;
+  }
+
+  if ((unvalidatedVal < (long long)min) || (unvalidatedVal > (long long)max)) {
+    std::cout << errorMessage;
+    return false;
+  }
+
+  map.erase(identifier);
+  output = (uint16_t)unvalidatedVal;
+  return true;
+}
+
+// Simple validators for RSA keys to ensure they at least conform to the
+// expected format.
+const size_t rsaPublicKeyHexadecimalLength = 584;
+
+static bool validateRSAPublicKey (const std::string& key) {
+  return (key.length() == rsaPublicKeyHexadecimalLength)
+      && (std::regex_match(key, std::regex("[0-9A-Fa-f]+")));
+}
+
+static bool validateRSAPrivateKey (const std::string& key) {
+
+  // Note we do not verify the length of RSA private keys because their length
+  // actually seems to vary a little in the output; it hovers around 2430
+  // characters but often does not exactly match that number.
+
+  return std::regex_match(key, std::regex("[0-9A-Fa-f]+"));
+}
+
+// Checks that there are entries for all identifiers given in entries in the
+// variable map produced by parsing the keyfile, and prints errors if there
+// are not.
+static bool expectEntries(const std::vector<std::string>& entries,
+                          const std::unordered_map<std::string, std::string>
+                            map,
+                          const std::string& filename,
+                          const std::unordered_map<std::string, size_t>
+                            lineNumbers) {
+  for (auto entry: entries) {
+    if (map.count(entry) < 1) {
+      if (lineNumbers.count(entry) < 1) {
+        std::cout << filename << ": Missing assignment for required parameter: "
+          << entry << ".\n";
+      } else {
+        std::cout << filename << ": line " << lineNumbers.at(entry)
+          << ": expected string value for " << entry << ", found list.\n";
+      }
+      return false;
+    }
+  }
+  return true;
+}
+
+// Handles reading all the public configuration for each of the four
+// cryptosystems we expect to read in.
+static bool deserializeCryptosystemPublicConfiguration(std::unique_ptr<
+                                                         Cryptosystem>&
+                                                         cryptosystem,
```

Comment:
That does sound like a good idea. I have restyled this; the change should be reflected in the next round of revisions I push.

I have also decided to make a similar restyling of `parseReplicaKeyfile`, `validateFundamentalFields`, `validateConfigStructIntegrity`, and `testThresholdCryptosystem`, which I thought also featured an unnecessary amount of long parameters wrapping to new lines because of how far to the right the parameter lists begin after starting them on the same line as the function names. Please feel free to inform me if any of these changes are unwarranted in these cases.